tests/integration/tokenomics/token_logic_modules/tlm_processor_suite_test.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/pokt-network/poktroll/testutil/proof"
+	"github.com/pokt-network/poktroll/testutil/sample"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	tlm "github.com/pokt-network/poktroll/x/tokenomics/token_logic_module"
+)
+
+type tlmProcessorTestSuite struct {
+	suite.Suite
+
+	ctx     context.Context
+	keepers testkeeper.TokenomicsModuleKeepers
+
+	service  *sharedtypes.Service
+	app      *apptypes.Application
+	supplier *sharedtypes.Supplier
+
+	proposerConsAddr cosmostypes.ConsAddress
+	sourceOwnerBech32,
+	foundationBech32 string
+
+	expectedSettledResults,
+	expectedExpiredResults tlm.PendingSettlementResults
+	expectedSettlementState *settlementState
+}
+
+// settlementState holds the expected post-settlement app stake and rewardee balances.
+type settlementState struct {
+	appStake             *cosmostypes.Coin
+	appBalance           *cosmostypes.Coin
+	supplierOwnerBalance *cosmostypes.Coin
+	proposerBalance      *cosmostypes.Coin
+	foundationBalance    *cosmostypes.Coin
+	sourceOwnerBalance   *cosmostypes.Coin
+}
+
+func init() {
+	cmd.InitSDKConfig()
+}
+
+func TestTLMProcessorTestSuite(t *testing.T) {
+	suite.Run(t, new(tlmProcessorTestSuite))
+}
+
+// SetupTest generates and sets all rewardee addresses on the suite, and
+// set a service, application, and supplier on the suite.
+func (s *tlmProcessorTestSuite) SetupTest() {
+	s.foundationBech32 = sample.AccAddress()
+	s.sourceOwnerBech32 = sample.AccAddress()
+	s.proposerConsAddr = sample.ConsAddress()
+
+	s.service = &sharedtypes.Service{
+		Id:                   "svc1",
+		ComputeUnitsPerRelay: 1,
+		OwnerAddress:         s.sourceOwnerBech32,
+	}
+
+	appStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, math.MaxInt64)
+	s.app = &apptypes.Application{
+		Address: sample.AccAddress(),
+		Stake:   &appStake,
+		ServiceConfigs: []*sharedtypes.ApplicationServiceConfig{
+			{ServiceId: s.service.GetId()},
+		},
+	}
+
+	supplierBech32 := sample.AccAddress()
+	s.supplier = &sharedtypes.Supplier{
+		OwnerAddress:    supplierBech32,
+		OperatorAddress: supplierBech32,
+		Stake:           &suppliertypes.DefaultMinStake,
+		Services: []*sharedtypes.SupplierServiceConfig{
+			{
+				ServiceId: s.service.GetId(),
+				RevShare: []*sharedtypes.ServiceRevenueShare{
+					{
+						Address:            supplierBech32,
+						RevSharePercentage: 50,
+					},
+				},
+			},
+		},
+		ServicesActivationHeightsMap: map[string]uint64{
+			s.service.GetId(): 0,
+		},
+	}
+}
+
+// getProofParams returns the default proof params with a high proof requirement threshold
+// and no proof request probability such that no claims require a proof.
+func (s *tlmProcessorTestSuite) getProofParams() *prooftypes.Params {
```

Comment:
Ditto elsewhere. Balance  betwen being too explicit but calling out that it's opinionaed.


```suggestion
func (s *tlmProcessorTestSuite) getDefaultProofParams() *prooftypes.Params {
```

Reply:
I disagree with this. I would be more inclined toward something like `getTestProofParams()` but perfer the current naming. Can you say more about your rationale and/or how strongly you feel about this?